ddl/ddl.go

```diff
@@ -212,6 +212,50 @@ type ddl struct {
 	// used in the concurrency ddl.
 	reorgWorkerPool      *workerPool
 	generalDDLWorkerPool *workerPool
+	// get notification if any DDL coming.
+	ddlJobCh chan struct{}
+	// recording the running jobs.
+	runningJobs struct {
+		sync.RWMutex
+		runningJobMap map[int64]struct{}
+	}
+	// used for build fetch DDL job SQL.
+	runningOrBlockedIDs []string
+}
+
+// waitSchemaSyncedController is to control whether to waitSchemaSynced or not.
+type waitSchemaSyncedController struct {
+	mu  sync.RWMutex
+	job map[int64]struct{}
+
+	// true if this node is elected to the DDL owner, we should wait 2 * lease before it runs the first DDL job.
+	once *atomicutil.Bool
+}
+
+func newWaitSchemaSyncedController() *waitSchemaSyncedController {
+	return &waitSchemaSyncedController{
+		job:  make(map[int64]struct{}, 16),
+		once: atomicutil.NewBool(true),
+	}
+}
+
+func (w *waitSchemaSyncedController) needSync(job *model.Job) {
```

Comment:
needSync is not a good name.

Reply:
I have changed it in the newest commit, PTAL